packages/ui/src/modal/buy/BuyModal.tsx

```diff
@@ -93,11 +95,17 @@ export function BuyModal({
     openState ? openState[0] : false,
     openState
   )
-  const { chain: activeChain } = useNetwork()
+
+  const { chains, chain } = useNetwork()
+
+  const activeChain = chainId
+    ? chains.find((chain) => chain.id === chainId) || chain
+    : chain
 
```

Comment:
One thing missing here is that we need to now check the chain when the buy button is clicked and prompt the switch network message.

Reply:
Let's also add the chainId to the /modal/buy page in the demo app.